File: implementations/java/src/main/java/com/webweavex/Validate.java
package com.webweavex;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;

import java.io.*;
import java.nio.file.*;
import java.util.*;

public class Validate {
    
    private static final Gson gson = new GsonBuilder()
        .setPrettyPrinting()
        .create();
    private static final JsonParser jsonParser = new JsonParser();
    
    public static void main(String[] args) throws Exception {
        String testCasesPath = "../../core/test_cases/test_cases.json";
        String outputDir = "../../test_output/java/";
        
        Path testCasesFile = Paths.get(testCasesPath).toAbsolutePath();
        Path outputPath = Paths.get(outputDir).toAbsolutePath();
        
        Files.createDirectories(outputPath);
        
        String jsonContent = new String(Files.readAllBytes(testCasesFile));
        JsonArray testCasesArray = jsonParser.parse(jsonContent).getAsJsonArray();
        
        List<String> testCaseNames = new ArrayList<>();
        
        WebWeaveX wx = new WebWeaveX();
        
        System.out.println("Exporting Java outputs...");
        System.out.println("==================================================");
        
        for (int i = 0; i < testCasesArray.size(); i++) {
            JsonObject tc = testCasesArray.get(i).getAsJsonObject();
            String name = tc.get("name").getAsString();
            String inputText = tc.get("input").getAsString();
            
            testCaseNames.add(name);
            
            System.out.println("Processing: " + name);
            
            Map<String, Object> result = wx.extract(inputText);
            String outputJson = gson.toJson(result);
            
            Path outputFile = outputPath.resolve(name + ".json");
            Files.write(outputFile, outputJson.getBytes());
            
            System.out.println("  Saved: " + outputFile);
        }
        
        System.out.println("==================================================");
        System.out.println("Exported " + testCasesArray.size() + " test cases to " + outputPath);
        
        JsonObject manifest = new JsonObject();
        manifest.addProperty("language", "java");
        JsonArray namesArray = new JsonArray();
        for (String n : testCaseNames) {
            namesArray.add(n);
        }
        manifest.add("test_cases", namesArray);
        Files.write(outputPath.resolve("manifest.json"), 
            gson.toJson(manifest).getBytes());
        System.out.println("Manifest: " + outputPath.resolve("manifest.json"));
    }
}

File: implementations/kotlin/src/main/kotlin/com/webweavex/Validate.kt
package com.webweavex

import com.google.gson.GsonBuilder
import com.google.gson.reflect.TypeToken
import java.io.File

object Validate {
    private val gson = GsonBuilder().setPrettyPrinting().create()
    private val webWeaveX = WebWeaveX()

    private const val TEST_CASES_PATH = "../../core/test_cases/test_cases.json"
    private const val OUTPUT_DIR = "../../test_output/kotlin"

    fun exportKotlinOutputs() {
        val testCasesFile = File(TEST_CASES_PATH)
        if (!testCasesFile.exists()) {
            println("Test cases not found: $TEST_CASES_PATH")
            System.exit(1)
        }

        val outputDir = File(OUTPUT_DIR)
        outputDir.mkdirs()

        val json = testCasesFile.readText()
        val testCases: List<Map<String, String>> = gson.fromJson(
            json,
            object : TypeToken<List<Map<String, String>>>() {}.type
        )

        println("Exporting Kotlin outputs...")
        println("=".repeat(50))

        val testCaseNames = mutableListOf<String>()

        for (tc in testCases) {
            val name = tc["name"] ?: continue
            val inputText = tc["input"] ?: ""

            println("Processing: $name")

            val result = webWeaveX.extract(inputText)
            val outputPath = File(OUTPUT_DIR, "$name.json")
            outputPath.writeText(gson.toJson(result))

            println("  Saved: ${outputPath.absolutePath}")
            testCaseNames.add(name)
        }

        println("=".repeat(50))
        println("Exported ${testCases.size} test cases to $OUTPUT_DIR")

        val manifestPath = File(OUTPUT_DIR, "manifest.json")
        val manifest = mapOf(
            "language" to "kotlin",
            "test_cases" to testCaseNames
        )
        manifestPath.writeText(gson.toJson(manifest))
        println("Manifest: ${manifestPath.absolutePath}")
    }
}

fun main() {
    Validate.exportKotlinOutputs()
}
